drivers/node/aws/aws.go

```diff
@@ -0,0 +1,119 @@
+package aws
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	aws_pkg "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/portworx/torpedo/drivers/node"
+)
+
+const (
+	// DriverName is the name of the aws driver
+	DriverName = "aws"
+)
+
+type aws struct {
+	node.Driver
+	session     *session.Session
+	credentials *credentials.Credentials
+	config      *aws_pkg.Config
+	region      string
+	svc         *ec2.EC2
+	instances   []*ec2.Instance
+}
+
+func (aws *aws) String() string {
+	return DriverName
+}
+
+func (aws *aws) Init(sched string) error {
+	sess := session.Must(session.NewSession())
+	aws.session = sess
+	creds := credentials.NewEnvCredentials()
+	aws.credentials = creds
+	aws.region = os.Getenv("AWS_REGION")
+	if aws.region == "" {
+		return fmt.Errorf("Env AWS_REGION not found")
+	}
+	config := &aws_pkg.Config{Region: aws_pkg.String(aws.region)}
+	config.WithCredentials(creds)
+	aws.config = config
+	svc := ec2.New(sess, config)
+	aws.svc = svc
+	instances := aws.getAllInstances()
+	aws.instances = instances
+	return nil
+}
+
+func (aws *aws) TestConnection(n node.Node, options node.TestConectionOpts) error {
+	return nil
+}
+
+func (aws *aws) RebootNode(n node.Node, options node.RebootNodeOpts) error {
+	var err error
+	instanceID, err := aws.getNodeIDByPrivAddr(n)
+	if err != nil {
+		return &node.ErrFailedToRebootNode{
+			Node:  n,
+			Cause: fmt.Sprintf("failed to get instance ID due to: %v", err),
+		}
+	}
+	//Reboot the instance by its InstanceID
+	rebootInput := &ec2.RebootInstancesInput{
+		InstanceIds: []*string{
+			aws_pkg.String(instanceID),
+		},
+	}
+	_, err = aws.svc.RebootInstances(rebootInput)
+	if err != nil {
+		return &node.ErrFailedToRebootNode{
+			Node:  n,
+			Cause: fmt.Sprintf("failed to reboot instance due to: %v", err),
+		}
+	}
+	return nil
+}
+
+func (aws *aws) ShutdownNode(n node.Node, options node.ShutdownNodeOpts) error {
+	return nil
+}
+
+func (aws *aws) getAllInstances() []*ec2.Instance {
+	params := &ec2.DescribeInstancesInput{}
+	resp, err := aws.svc.DescribeInstances(params)
+	if err != nil {
+		logrus.Errorf("there was an error listing instances in %s. Error: %q", aws.region, err.Error())
+		logrus.Fatal(err.Error())
+	}
+	reservations := resp.Reservations
+	instances := []*ec2.Instance{}
+	for _, resv := range reservations {
+		for _, ins := range resv.Instances {
+			instances = append(instances, ins)
+		}
+	}
+	return instances
+}
+
+func (aws *aws) getNodeIDByPrivAddr(n node.Node) (string, error) {
+	for _, i := range aws.instances {
+		for _, addr := range n.Addresses {
+			if aws_pkg.StringValue(i.PrivateIpAddress) == addr {
+				return aws_pkg.StringValue(i.InstanceId), nil
+			}
+		}
+	}
+	return "", fmt.Errorf("Failed to get instanceID by privateIP")
```

Comment:
Include the node name (`n.Name`) here in the Errorf.

Reply:
fixed as
return "", fmt.Errorf("Failed to get instanceID of %s by privateIP", n.Name)